src/main/java/minijava/ir/optimize/CommonSubexpressionElimination.java

```diff
@@ -315,10 +360,12 @@ private void addHashedNode(HashedNode hashed) {
    * been computed yet (or can't be, e.g. Phis) then we return Optional.empty().
    */
   private Optional<Integer> hashWithSalt(int salt, Node... preds) {
-    // Java really needs `mapM`. /jk
-    Object[] hashedPreds = Seq.of(preds).map(hashes::get).toArray(Object[]::new);
-    if (Seq.of(hashedPreds).contains(null)) {
-      return Optional.empty();
+    Object[] hashedPreds = new Object[preds.length];
+    for (int i = 0; i < preds.length; ++i) {
+      hashedPreds[i] = hashes.get(preds[i]);
+      if (hashedPreds[i] == null) {
+        return Optional.empty();
+      }
     }
     return Optional.of(salt ^ Objects.hash(hashedPreds));
```

Comment:
You're right, I thought about it when I hit that line. But that would mean we'd have to wrap the entire call in a new array, something I didn't really want to do. We should be good anyways because conflicts are resolved by the `ExtensionalEqualityComparator`.